Foundation/NSNumber.swift

```diff
@@ -226,12 +226,43 @@ open class NSNumber : NSValue {
         case let other as Bool:
             return boolValue == other
         case let other as NSNumber:
-            return CFEqual(_cfObject, other._cfObject)
+            // CFEqual() requires both values to be the same type.
```

Comment:
I took a look at the Darwin implementation here. It defers to compare for all of this. A few notes for comparison:

For doubles d1/d2:
1. if d1 && d2 are NaN -> ordered same
2. If d2 is negative (including -0), we put nan after it; otherwise before it
3. If d1 is negative (including -0), we put nan after it; otherwise before it

For 'Q' type: call unsignedLongLongValue and compare that

For everything else: call longLongValue and compare that result.

Reply:
`@YES.longLongValue == 1`
`@NO.longLongValue == 0`

`NSNumber(value: 1.1) != NSNumber(value: Int64(1))`

Reply:
The NSNumber code certainly handles float/double differently - it could be a consequence of bridging which is causing `Float(1.1)` to compare equal to `true`. What's your exact test case so I can check?

Reply:
I must have done something wrong in my initial testing as `compare(other) == .orderedSame` seems to work ok now!